### PR TITLE
assignment 01 submission Zac Zhuo Zhang

### PR DESCRIPTION
**Pull request by ZacZhangzhuo**
1 file changed, 1 insertion(+), 0 deletions(-) create mode 100644 lecture_02/assignment_01/ZacZhuoZhang/assignment_01.py
image

### CODE REVIEW

File: lecture_02/assignment_01/ZacZhuoZhang/assignment_01.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+mesh_projected = mesh.transformed(P)
+# outTemp = mesh_projected
+
+# Create artists
+artist1 = Artist(box)
+artist2 = Artist(mesh_projected)
+
+# Draw and all to a list
+# artist1.draw()
+# artist2.draw_edges(color="#00ff00")
```
Comment (gonzalocasas) [inline lecture_02/assignment_01/ZacZhuoZhang/assignment_01.py:33]: Without drawing, there's no output visible. Please uncomment these lines.

**Review by gonzalocasas — CHANGES_REQUESTED** [review]
Please address the inlined comments.

Also, the pull request template asks for a screenshot of the solution, but that refers to the output in Rhino, not a screenshot of the files or the commit output of git ;)

File: lecture_02/assignment_01/ZacZhuoZhang/assignment_01.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Define a Frame, which is not in the origin and a bit tilted to the world frame
+frame = Frame([1, 1, 1], [1, 1, 0], [0,1, 1])
+
+# Create a Box with that frame
+box = Box(frame, 1, 1, 1)
+outTemp = box
+
+# Create a Projection (can be orthogonal, parallel or perspective)
+P  = Projection.from_plane_and_direction(Plane.worldXY(), Vector(math.sin(θ)*math.cos(φ), math.sin(θ)*math.sin(φ),math.cos(θ)))
```
[the remark this answers is not in the document]
Comment (ZacZhangzhuo) [reply lecture_02/assignment_01/ZacZhuoZhang/assignment_01.py:18]: Sorry for the confusion. The idea is that we can input `θ` and `φ` from (r, θ, φ) coordination so we can create the 'sun' from any angle, and get the cube's shadow. I have changed `θ` -> `a` and `φ` -> `b` in the next version.

File: lecture_02/assignment_01/ZacZhuoZhang/assignment_01.py (hunk 1 above)
Comment (ZacZhangzhuo) [reply lecture_02/assignment_01/ZacZhuoZhang/assignment_01.py]: Thanks. I usually open a template GH file with a prefab python component and an `outTemp` output. The python component connects to a `Read File` input and a `Compas Draw` output so I can code directly without GH operations. Anyway I have fix it.

**Comment by ZacZhangzhuo** [comment]
Thanks of the comments!

image

**Review by gonzalocasas — APPROVED** [review]
Nice! 👍

**Pull request merged**